scripts/mirror.js

```diff
@@ -0,0 +1,617 @@
+      modify,
+    );
+  } else {
+    if (data[rootPath].constructor == Object || Array.isArray(data[rootPath])) {
+      newData = doSetFeature(data, newData, rootPath, browser, source, modify);
+    }
+  }
+
+  return newData;
+};
+
+/**
+ * @param {Identifier} data
+ * @param {string} browser
+ * @param {string} source
+ * @param {string} modify
+ * @returns {Identifier}
+ */
+const setFeatureRecursive = (data, browser, source, modify) => {
+  let newData = Object.assign({}, data);
+
+  for (let i in data) {
+    if (!!data[i] && typeof data[i] == 'object' && i !== '__compat') {
+      newData[i] = data[i];
+      if (data[i].__compat) {
+        doSetFeature(data, newData, i, browser, source, modify);
+      }
+      setFeatureRecursive(data[i], browser, source, modify);
+    }
+  }
+
+  return newData;
+};
+
+/**
+ * @param {string} browser
+ * @param {string} filepath
+ * @param {string} source
+ * @param {string} modify
+ * @returns {boolean}
+ */
+function mirrorDataByFile(browser, filepath, source, modify) {
+  let file = filepath;
+  if (file.indexOf(__dirname) !== 0) {
+    file = path.resolve(__dirname, '..', file);
+  }
+
+  if (!fs.existsSync(file)) {
+    return false;
+  }
+
+  if (fs.statSync(file).isFile()) {
+    if (path.extname(file) === '.json') {
+      let data = require(file);
+      let newData = setFeatureRecursive(data, browser, source, modify);
+
+      fs.writeFileSync(file, JSON.stringify(newData, null, 2) + '\n', 'utf-8');
+    }
+  } else if (fs.statSync(file).isDirectory()) {
+    const subFiles = fs.readdirSync(file).map(subfile => {
+      return path.join(file, subfile);
+    });
+
+    for (let subfile of subFiles) {
+      mirrorDataByFile(browser, subfile, source, modify);
+    }
+  }
+
+  return true;
+}
+
+/**
+ * Allows mirroring by feature ID (e.g. "html.elements.a")
+ *
+ * Note that this assumes a predictable file structure
+ * which BCD doesn't have right now. (issue #3617)
+ * For example, even if "html.elements.input.input-button"
+ * is a valid query, it will fail here, because the file structure
+ * for input-button isn't consistent with the rest right now.
+ *
+ * @param {string} browser
+ * @param {string} featureIdent
+ * @param {string} source
+ * @param {string} modify
+ * @returns {boolean}
+ */
+const mirrorDataByFeature = (browser, featureIdent, source, modify) => {
+  let filepath = path.resolve(__dirname, '..');
+  let feature = featureIdent.split('.');
+  let found = false;
+
+  for (let depth = 0; depth < feature.length; depth++) {
+    filepath = path.resolve(filepath, feature[depth]);
+    const filename = filepath + '.json';
+    if (fs.existsSync(filename) && fs.statSync(filename).isFile()) {
+      filepath = filename;
+      found = true;
+      break;
+    }
+  }
+
+  if (!found) {
+    console.error(`Could not find file for ${featureIdent}!`);
+    return false;
+  }
+
+  let data = require(filepath);
+  let newData = setFeature(data, feature, browser, source, modify);
```

Comment:
Now that we keep `mirrorDataByFeature` (and I agree, it is likely very useful!), we should document that it doesn't work recursively. Do you think we should mention that in the contributing.md, too?
`html.elements.button.*` could be recursive syntax, but the main BCD querying doesn't support that either.